src/main/java/com/vmware/vipclient/i18n/messages/service/LocaleService.java

```diff
@@ -116,11 +124,14 @@ public LocaleService(LocaleDTO dto) {
 
     private Map<String, String> getSupportedLanguagesFromDS(String locale, ListIterator<DataSourceEnum> msgSourceQueueIter) {
         Map<String, String> dispMap = new HashMap<String, String>();
-        if (!msgSourceQueueIter.hasNext())
+        if (!msgSourceQueueIter.hasNext()) {
+            logger.error(FormatUtils.format(ConstantsMsg.GET_LANGUAGES_FAILED_ALL));
             return dispMap;
+        }
         DataSourceEnum dataSource = (DataSourceEnum) msgSourceQueueIter.next();
         dispMap = dataSource.createLocaleOpt(dto).getSupportedLanguages(locale);
         if (dispMap == null || dispMap.isEmpty()) {
+            logger.debug(FormatUtils.format(ConstantsMsg.GET_LANGUAGES_FAILED, dataSource.toString()));
```

Comment:
Suggest log in this way. FromatUtils is heavy and unnecessary.

```suggestion
            logger.debug(ConstantsMsg.GET_LANGUAGES_FAILED, dataSource);
```